code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -109,25 +109,32 @@
 
 /obj/machinery/atmospherics/components/unary/thermomachine/process_atmos()
 	..()
-	if(!on || !nodes[1])
+	if(is_operational)
+		if(!on || !nodes[1])
+			return
+		var/datum/gas_mixture/air_contents = airs[1]
+
+		var/air_heat_capacity = air_contents.heat_capacity()
+		var/combined_heat_capacity = heat_capacity + air_heat_capacity
+		var/old_temperature = air_contents.temperature
+
+		if(combined_heat_capacity > 0)
+			var/combined_energy = heat_capacity * target_temperature + air_heat_capacity * air_contents.temperature
+			air_contents.temperature = combined_energy/combined_heat_capacity
+
+		var/temperature_delta= abs(old_temperature - air_contents.temperature)
+		if(temperature_delta > 1)
+			active_power_usage = (heat_capacity * temperature_delta) / 10 + idle_power_usage
+			update_parents()
+		else
+			active_power_usage = idle_power_usage
+		return 1
```

Comment:
What's the 1 for? Is it in place of `TRUE`, or is this meant to be a number?

Reply:
it was there before hand, the whole procs parent defaults to FALSE, setting to true actually shuts it off from what i can tell. in any case, i refuse to touch the code for atmos, its outside the scope of this PR, im only here to fix the power issue

Reply:
I'm just asking you to change it to `return TRUE` while you're here 😛